internal/zinc-core/src/main/scala/sbt/internal/inc/Incremental.scala

```diff
@@ -366,10 +366,15 @@ object Incremental {
     if (earlyOutput.isDefined || isPickleWrite) {
       val idx = scalacOptions.indexOf("-Ypickle-write")
       val p =
-        if (scalacOptions.size <= idx + 1) None
+        if (!isPickleWrite || scalacOptions.size <= idx + 1) None
         else Some(Paths.get(scalacOptions(idx + 1)))
       (p, earlyJar) match {
-        case (None, _)            => log.warn(s"-Ypickle-write is specified but <path> is not?")
+        case (None, _) if !isPickleWrite =>
+          if (isPickleWrite) log.warn(s"expected -Ypickle-write <path> but <path> is missing")
```

Comment:
how would this ever be true?

Reply:
This is to account for someone passing `-Ypickle-write` as scalac options instead of `-Ypickle-write /some/path`

Reply:
Opps. I see the if above.